addons/godot-next/nodes/behavior_collection.gd

```diff
@@ -0,0 +1,165 @@
+	"_input" : {},
+	"_unhandled_input" : {},
+	"_unhandled_key_input" : {}
+}
+
+##### NOTIFICATIONS #####
+
+func _set(p_property: String, p_value) -> bool:
+	if not Engine.editor_hint:
+		return false
+	if PropertyUtils.is_set_element(p_property):
+		return PropertyUtils.set_set_element(self, p_property, p_value)
+	return false
+
+func _get(p_property: String):
+	if not Engine.editor_hint:
+		return null
+	if PropertyUtils.is_set_element(p_property):
+		return PropertyUtils.get_set_element(self, p_property)
+	return null
+
+func _get_property_list() -> Array:
+	var list := []
+	list += PropertyUtils.generate_resource_set(self, "_behaviors", Behavior)
+	return list
+
+func _ready() -> void:	
+	for behavior in _behaviors.values():
+		_initialize_behavior(behavior)
+	
+	for behavior in _callbacks["_ready"]:
+		behavior._ready()
+
+func _process(delta: float) -> void:
+	for behavior in _callbacks["_process"]:
+		if behavior.enabled:
+			behavior._process(delta)
+
+func _physics_process(delta: float) -> void:
+	for behavior in _callbacks["_physics_process"]:
+		if behavior.enabled:
+			behavior._process(delta)
+
+func _input(event: InputEvent) -> void:
+	for behavior in _callbacks["_input"]:
+		if behavior.enabled:
+			behavior._input(event)
+
+func _unhandled_input(event: InputEvent) -> void:
+	for behavior in _callbacks["_unhandled_input"]:
+		if behavior.enabled:
+			behavior._unhandled_input(event)
+
+func _unhandled_key_input(event: InputEventKey) -> void:
+	for behavior in _callbacks["_unhandled_key_input"]:
+		if behavior.enabled:
+			behavior._process(event)
+
+##### VIRTUALS #####
+
+func _add_element(p_variable: String, p_script: Script) -> void:
+	var class_type = ClassType.new(p_script)
+	var key = class_type.get_script_class()
+	get(p_variable)[key] = p_script.new()
+
+##### PUBLIC METHODS #####
+
+func add_behavior(p_type: Script) -> Behavior:
+	var ct = ClassType.new(p_type)
+	if not ct.type_is(Behavior):
+		return null
+	if has_behavior(p_type):
+		return get_behavior(p_type)
+		
+	var behavior: Behavior = p_type.new()
+	var behavior_name: String = ct.get_script_class()
+	
+	_behaviors[behavior_name] = behavior	
+	_initialize_behavior(behavior)
+	
+	return behavior
+
+func get_behavior(p_type: Script) -> Behavior:
+	var ct = ClassType.new(p_type)
+	var type_name: String = ct.get_script_class()
+	if _behaviors.has(type_name):
+		return _behaviors[type_name]
+	return null
+
+func has_behavior(p_type: Script) -> bool:
+	var ct = ClassType.new(p_type)
+	var type_name: String = ct.get_script_class()
+	return _behaviors.has(type_name)
+
+func remove_behavior(p_type: Script) -> bool:
+	var ct = ClassType.new(p_type)
+	var type_name: String = ct.get_script_class()
+	if not _behaviors.has(type_name):
+		return false
+	var behavior = _behaviors[type_name]
+	#warning-ignore:return_value_discarded
+	_behaviors.erase(type_name)
+	_remove_from_callbacks(behavior)
+	behavior.disconnect("script_changed", self, "_refresh_callbacks", [behavior])
```

Comment:
Afaik, objects like this are automatically disconnected from all signals if their memory is freed, so I don't think you need to manually disconnect it just before freeing.

Also, FYI, `disconnect()` only takes three parameters. It does not require the other details as the signal name, callback object, and callback method are all that is needed for a primary key to identify the connection.

Reply:
Remove the whole `disconnect` line!